cardinal/ecs/guid.go

```diff
@@ -0,0 +1,81 @@
+package ecs
+
+import (
+	"errors"
+
+	"github.com/cometbft/cometbft/crypto"
+)
+
+const numOfCharactersInGUID = 24
+
+// GUIDComponent is a unique game identification string. It is generated randomly on tick 0 and
+// will never change.
+type GUIDComponent struct {
+	GUID string `json:"guid"`
+}
+
+func (GUIDComponent) Name() string {
+	return "GameUniqueIDComponent"
```

Comment:
I would be in favour of changing this to "GUIDComponent", but I suppose either way works